src/qililab/calibration/calibration_controller.py

```diff
@@ -642,80 +652,127 @@ def get_last_fidelities(self) -> pd.DataFrame:
-                and node.previous_timestamp is not None
-                and "platform_parameters" in node.output_parameters
-            ):
+            if "platform_parameters" in node.output_parameters:
                 for bus, _, parameter, _ in node.output_parameters["platform_parameters"]:
                     bus_list = str(bus).split("_")
                     bus = "_".join([x for x in bus_list if not any(char.isdigit() for char in x)])
 
                     if f"{str(parameter)}_{bus}" not in col:
                         col.append(f"{str(parameter)}_{bus}")
 
-        df = pd.DataFrame("-", idx, col)
-        df.index.name = "qubit"
+        return idx, col
 
-        return df
+    @staticmethod
+    def _fill_qubits_columns(node: CalibrationNode, qubit: str, df: pd.DataFrame) -> pd.DataFrame:
+        """Fills the qubit tables columns with the fidelities or parameters values.
 
-    def get_qubits_table(self) -> pd.DataFrame:
-        """Retrieves the last updated fidelities of the graph in a qubit table.
+        Args:
+            node (CalibrationNode): Node currently being filled.
+            qubit (str): qubit being filled, corresponding to the node.
+            df (pd.DataFrame): Dataframe to fill.
 
         Returns:
-            pd.DataFrame: Table where columns are each fidelity or parameter, and rows each qubit.
+            pd.DataFrame: Filled dataframe.
         """
-        df = self._create_empty_dataframe()
-
-        for node in self.node_sequence.values():
-            qubit_list = node.node_id.split("_")
-            qubit = "_".join(
-                [i for i in qubit_list if any(char == "q" for char in i) and any(char.isdigit() for char in i)]
-            ).replace("q", "-")[1:]
-
-            if (
-                node.output_parameters is not None
-                and node.previous_timestamp is not None
-                and "fidelities" in node.output_parameters
-            ):
+        if node.output_parameters is not None and node.previous_timestamp is not None:
+            if "fidelities" in node.output_parameters:
                 for _, fidelity, value in node.output_parameters["fidelities"]:
                     df[fidelity][qubit] = value
 
-            if (
-                node.output_parameters is not None
-                and node.previous_timestamp is not None
-                and "platform_parameters" in node.output_parameters
-            ):
+            if "platform_parameters" in node.output_parameters:
                 for bus, _, parameter, value in node.output_parameters["platform_parameters"]:
                     bus_list = str(bus).split("_")
                     bus = "_".join([x for x in bus_list if not any(char.isdigit() for char in x)])
                     df[f"{str(parameter)}_{bus}"][qubit] = value
 
-        # Reorder fidelities to the front of the dataframe:
+        return df
+
+    @staticmethod
+    def _reorder_fidelities(df: pd.DataFrame) -> pd.DataFrame:
```

Comment:
Is this for display purposes or we need them at front for the cloud team database format?

Reply:
Display purposes 👌

Reply:
Cool, thanks!